src/main/java/frc/robot/superstructure/Superstructure.java

```diff
@@ -127,71 +178,125 @@ public Command speaker() {
+				new ParallelCommandGroup(
+					elevatorRollerStateHandler.setState(ElevatorRollerState.STOP),
+					elevatorStatesHandler.setState(ElevatorStates.IDLE)
+				)
+			),
 			flywheelStateHandler.setState(FlywheelState.DEFAULT),
-			elevatorRollerStateHandler.setState(ElevatorRollerState.AMP),
-			funnelStateHandler.setState(FunnelState.AMP),
+			intakeStatesHandler.setState(IntakeStates.STOP),
 			pivotStateHandler.setState(PivotState.UP),
-			elevatorStatesHandler.setState(ElevatorStates.AMP)
+			funnelStateHandler.setState(FunnelState.STOP)
 		);
 	}
 
 	public Command transferShooterElevator() {
 		return new ParallelCommandGroup(
+			setCurrentStateValue(RobotState.TRANSFER_SHOOTER_ELEVATOR),
+			new SequentialCommandGroup(
+				new ParallelCommandGroup(
+					intakeStatesHandler.setState(IntakeStates.STOP),
+					funnelStateHandler.setState(FunnelState.STOP)
+				),//.until(isReadyToTransfer)
+				new ParallelCommandGroup(
+					intakeStatesHandler.setState(IntakeStates.SHOOTER_TO_ELEVATOR),
+					funnelStateHandler.setState(FunnelState.SHOOTER_TO_ELEVATOR)
+				).until(this::isNoteInElevatorRoller),
+				new ParallelCommandGroup(
+					intakeStatesHandler.setState(IntakeStates.STOP),
+					funnelStateHandler.setState(FunnelState.STOP)
+				)
+			),
 			swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE),
+			new SequentialCommandGroup(
+				elevatorRollerStateHandler.setState(ElevatorRollerState.TRANSFER_TO_ELEVATOR).until(this::isNoteInElevatorRoller),
+				elevatorRollerStateHandler.setState(ElevatorRollerState.STOP)
+			),
 			flywheelStateHandler.setState(FlywheelState.DEFAULT),
-			elevatorRollerStateHandler.setState(ElevatorRollerState.TRANSFER_TO_ELEVATOR),
-			funnelStateHandler.setState(FunnelState.SHOOTER_TO_ELEVATOR),
-			intakeStatesHandler.setState(IntakeStates.SHOOTER_TO_ELEVATOR),
-			pivotStateHandler.setState(PivotState.UP),
-			elevatorStatesHandler.setState(ElevatorStates.IDLE)
+			elevatorStatesHandler.setState(ElevatorStates.IDLE),
+			pivotStateHandler.setState(PivotState.UP)
 		);
 	}
 
 	public Command transferElevatorShooter() {
 		return new ParallelCommandGroup(
+			setCurrentStateValue(RobotState.TRANSFER_ELEVATOR_SHOOTER),
+			new SequentialCommandGroup(
+				new ParallelCommandGroup(
+					intakeStatesHandler.setState(IntakeStates.STOP),
+					funnelStateHandler.setState(FunnelState.STOP)
+				),//.until(isReadyToTransfer)
+				new ParallelCommandGroup(
+					intakeStatesHandler.setState(IntakeStates.ELEVATOR_TO_SHOOTER),
+					funnelStateHandler.setState(FunnelState.NOTE_TO_SHOOTER)
+				).until(this::isNoteInShooter),
+				new ParallelCommandGroup(
+					intakeStatesHandler.setState(IntakeStates.STOP),
+					funnelStateHandler.setState(FunnelState.STOP)
+				)
+			),
 			swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE),
+			new SequentialCommandGroup(
+				elevatorRollerStateHandler.setState(ElevatorRollerState.TRANSFER_TO_ELEVATOR).until(() -> !this.isNoteInElevatorRoller()),
```

Comment:
Doesn't look quite right, wouldn't this eject the note? Shouldnt this be transfer from elevator?